### ISSUE TITLE
select breaks syntax highlighting in bash

### ISSUE THREAD

**Issue by DarioDarko** [vim/vim#14795]
### Steps to reproduce

set filetype to sh or bash
use 'select' as a command argument without surrounding quotes
watch the syntax highlighting break

example:

```
while true
do
	anycommand select something
done
```

i randomly discovered this because copyq has a select argument, that i use in a script

### Expected behaviour

'select' should not break the syntax highlighting if its used as an argument for a command. the syntax recognition logic should only trigger if its at the very beginning of the line, after && semicolon etc

### Version of Vim

9.1.0408-1

### Environment

EndeavourOS
Alacritty

**Comment by chrisbra** [comment]
I don't see this. Can you please verify using `vim --clean`
![grafik](https://github.com/vim/vim/assets/244927/0034b303-2120-46b5-bc39-fd050f34dce3)

**Comment by DarioDarko** [comment]
vim --clean

![image](https://github.com/vim/vim/assets/154679092/cc7440c8-0e8e-447e-bced-93e34bf12ce3)

**Comment by chrisbra** [comment]
Hm, did you use `bash` initially? I only see this when using `bash` in the shebang line. 
This seems to happen when the syntax script determines a bash script, because bash "knows" `select name [ in word ] ; do list ; done`, which the syntax script tries to match here. And since there is a missing `do` here, it matches the done with an error. You can workaround this by setting `:let b:is_posix=1`.

**Comment by DarioDarko** [comment]
yes normally i use bash, but for me this happens with sh as well

btw this does not only affect select. same with while, until, for, etc

**Comment by chrisbra** [comment]
> btw this does not only affect select. same with while, until, for, etc

Yeah, for the same reason, you are using reserved words which the syntax script tries to match against the expected syntax.

> :let b:is_posix=1

So does this help?

**Comment by DarioDarko** [comment]
> Yeah, for the same reason, you are using reserved words which the syntax script tries to match against the expected syntax.
yes, i understand the issue. wouldt it be pretty simple to fix that in the syntax script?

> So does this help?
unfortunately no
![image](https://github.com/vim/vim/assets/154679092/5be8e461-b669-43a0-aad0-53fcee4c4691)
